03-task-manager/starter/QuizAnswers2.txt

```diff
@@ -0,0 +1,16 @@
+In this lesson, you created a middleware function called asyncWrapper. Why?
+
+By using asyncWrapper, I can write asynchronous route handlers in a clean way without having to write a try-catch block again and again. 
+So, I can handle all errors in a centralized error handling middleware.
+
+Suppose that you want to make sure that both a status code and an error message are sent back to the user when they request the URL for a task that does not exist. Assume that you’ve created a CustomAPIError class and an error handler that references that class. Complete the code:
+const getTask = asyncWrapper(async (req, res, next) => {  
+  const { id: taskID } = req.params;  
+  const task = await Task.findOne({ _id: taskID });  
+  if (!task) {  
+    // your code here  
+    const err = new CustomAPIError('No task found with the given id', 404);
+    return next(err);
```

Comment:
Note (no changes required):
You have two options for this custom error. You can either call `next` with the error (which you've done) OR throw this error since it's being handled within the `asyncWrapper` middleware function! Which to use is personal preference. I like that throwing the error is cleaner, but it can be confusing as to where the error is being caught/handled.

```js
throw new CustomAPIError(`Task with ID ${taskID} not found`, 404);
```
vs
```js
return next(new CustomAPIError(`Task with ID ${taskID} not found`, 404));
```